### ISSUE TITLE
Getting original text (input) from dataloader or datasilo

### ISSUE THREAD

**Issue by gokberkyar** [deepset-ai/FARM#535]
**Question**
Getting original text (input) from dataloader or datasilo ?

Hi,

I am using cross-validation, ie DataSiloForCrossVal in strafied format. 
In short data is shuffled. 
 `evaluator.eval(model, return_preds_labels = True) 
`
Returns labels BUT I could not find a way to link preds to original input text for verification purposes. I want to create a df/csv/excel to individual check whether it is correct or not. 

Since I dont know this portion is test/dev/test. I cannot use inference mod, that would produce labels but i would contain results from train set as well. 

In short how to get original data after datasilo is constructed. 
**Additional context**
Add any other context or screenshots about the question (optional).

**Comment by Timoeller** [comment]
Hey @gokberkyar the functionality you want is not in FARM. Especially the datasilo for cross val is rather hidden from the user. In the normal datasilo there is the option to keep input text, but you would need completely change the data handling and indexing in the cross validator class. From a quick glance this change looks rather complicated.

An easy way to get predictions in combination with the input data is to use the inferencer:
Split your data into train, dev + test sets. Use the trainset during crossval for finding the best parameters. Then train a model normally (without crossval) with trainset and devset, save the model, load it with the inferencer and manually evaluate the predictions on the testset. Would that also be an option?

**Comment by gokberkyar** [comment]
Thanks a lot for quick reply.

I am planning to use crossvalidation from another library to create train/dev/test sets for each fold. Then as you mention I will use datasilo in a normal fashion, and merge the results. It would be great if all handled through FARM but anyways this is a wonderful tool!.
